### ISSUE TITLE
asyncUploadPhotos in BVBaseUGCSubmission.m never adds photos 

### ISSUE THREAD

**Issue by gannonmg** [bazaarvoice/bv-ios-sdk#29]
## The Problem
When I tried to use addPhotos to add UIImages to a review, the review would successfully go through, but images never would.

## Steps to Reproduce
In version 8.0.3 of the SDK, try to attach a photo to a review using the addPhoto function. The review will not show any photos once it is uploaded to the BV dashboard.

## Expected Behavior
I'd like to be able to trust that the BVSDK is actually going to submit my users' photos when they try to attach them

## Environment
Happens on all iOS and Xcode versions. I am currently using the source code of the SDK from the December 12, 2018 version 8.0.3 release. It seems to have originated from this Oct 12, 2018 refactor.

https://github.com/bazaarvoice/bv-ios-sdk/commit/95c9486e81d4d403e297eb6f49148cfa86c04318#diff-113a8673f2f79fa9a2152230ef9f824e

## Details
The issue happens around line 83 in BVBaseUGCSubmission.m. In that for loop, photos and captions are only added to photoURLs and photoCaptions if photoURL && photoCaption are not null (line ~84). However, in my experience, response.result.photo.caption is always null. I fixed this by using the reference to photo.photoCaption that is available in the scope of the loop, instead of hoping that the API sends back a photo caption that isn't nil.

## Link to BVSDK Logs
No logs. The issue does not cause a direct crash, instead it just results in certain UGC not appearing.

## Code To Reproduce Issue [ Very Good To Have ]
No code to reproduce (just try creating a review submission, adding photos, and then calling review.submit). But I will show you the quick fix I employed in the meantime.

Around line 82 in the asyncUploadPhotos function in BVBaseUGCSubmission.m:
          NSString *photoURL = response.result.photo.sizes.normalUrl;
//          NSString *photoCaption = response.result.photo.caption;     //always returns null!
          NSString *photoCaption = photo.photoCaption;     //In my case this is never nil, just maybe an empty string
          if (photoURL && photoCaption) {      //always fails if photoCaption is nil
            [photoURLs addObject:photoURL];
            [photoCaptions addObject:photo.photoCaption];
          }

In short, since photoCaption always seems to be nil, the photoURLs and Captions are enver added to the proper arrays for submission.
